In [2]:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.tree import DecisionTreeClassifier
from sklearn.metrics import roc_auc_score,accuracy_score
from sklearn.ensemble import BaggingClassifier

In [3]:
df= pd.read_csv("Bankruptcy.csv")

In [4]:
X = df.iloc[:,2:]
y = df.iloc[:,1]

In [5]:
# Create training and test sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.3, random_state=2022,stratify=y)

In [6]:
log_reg = LogisticRegression()
dtc =  DecisionTreeClassifier(random_state=2022)

In [8]:
model_bg = BaggingClassifier(base_estimator=log_reg,n_estimators=15,max_samples=X_train.shape[0],max_features=X_train.shape[1],random_state=2022)
model_bg.fit(X_train,y_train)
y_pred_prob = model_bg.predict_proba(X_test)[:,1]
print("roc_auc_score with log_reg  = ",roc_auc_score(y_test, y_pred_prob))

roc_auc_score with log_reg  =  0.82


C:\Users\Testbook\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(
C:\Users\Testbook\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/st

In [9]:
model_bg = BaggingClassifier(base_estimator=dtc,n_estimators=15,max_samples=X_train.shape[0],max_features=X_train.shape[1],random_state=2022)
model_bg.fit(X_train,y_train)
y_pred_prob = model_bg.predict_proba(X_test)[:,1]
print("roc_auc_score with dtc  = ",roc_auc_score(y_test, y_pred_prob))

roc_auc_score with dtc  =  0.90625


- roc_auc_score with LogisticRegression   =  0.82
- roc_auc_score with DecisionTreeClassifier  =  0.90625

From Above we can conclude that DecisionTree Classifier with Bagging is the Best Model for the given Data , Having roc_auc_score of  0.90625